boa/src/syntax/ast/node/statement_list.rs

```diff
@@ -62,3 +64,24 @@ impl fmt::Display for StatementList {
         self.display(f, 0)
     }
 }
+
+#[derive(Clone, Debug, Finalize, PartialEq)]
+pub struct RcStatementList(Rc<StatementList>);
+
+impl Deref for RcStatementList {
+    type Target = StatementList;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<StatementList> for RcStatementList {
+    #[inline]
+    fn from(statementlist: StatementList) -> Self {
+        Self(Rc::from(statementlist))
+    }
+}
+
+unsafe impl Trace for RcStatementList {
```

Comment:
It would be nice to add a comment on why it is `unsafe` why is this code ok with `SAFETY: `